core/vm/operations_acl.go

```diff
@@ -230,6 +230,10 @@ func makeSelfdestructGasFn(refundsEnabled bool) gasFunc {
 			// If the caller cannot afford the cost, this change will be rolled back
 			evm.StateDB.AddAddressToAccessList(address)
 			gas = params.ColdAccountAccessCostEIP2929
+
+			if contract.Gas < gas {
```

Comment:
According to the EIP https://eips.ethereum.org/EIPS/eip-7928

BALs must include the Addresses accessed without state changes, including: Beneficiary addresses for SELFDESTRUCT

It means the beneficiary address should be tracked in the BALs regardless of the gas is sufficient should this opcode.
If we short circuit here and without executing `if evm.StateDB.Empty(address) && evm.StateDB.GetBalance(contract.Address()).Sign() != 0 {`
how should be explicitly track it?